Index: linux-4.14.172/drivers/mtd/nand/nand_ids.c
===================================================================
--- linux-4.14.172.orig/drivers/mtd/nand/nand_ids.c
+++ linux-4.14.172/drivers/mtd/nand/nand_ids.c
@@ -54,6 +54,9 @@ struct nand_flash_dev nand_flash_ids[] =
 		{ .id = {0xad, 0xde, 0x94, 0xda, 0x74, 0xc4} },
 		  SZ_8K, SZ_8K, SZ_2M, NAND_NEED_SCRAMBLING, 6, 640,
 		  NAND_ECC_INFO(40, SZ_1K), 4 },
+	{"GD9FU1G8F3A 1G 3.3V 8-bit",
+	    { .id = {0xc8, 0xf1, 0x80, 0x19, 0x42} },
+	      SZ_2K, SZ_128, SZ_128K, 0, 5, 64, NAND_ECC_INFO(4, SZ_512) },
 
 	LEGACY_ID_NAND("NAND 4MiB 5V 8-bit",   0x6B, 4, SZ_8K, SP_OPTIONS),
 	LEGACY_ID_NAND("NAND 4MiB 3,3V 8-bit", 0xE3, 4, SZ_8K, SP_OPTIONS),
