Index: linux-4.14.172/arch/mips/ath79/mach-rb95x.c
===================================================================
--- linux-4.14.172.orig/arch/mips/ath79/mach-rb95x.c
+++ linux-4.14.172/arch/mips/ath79/mach-rb95x.c
@@ -33,6 +33,7 @@
 
 #include <asm/mach-ath79/ath79.h>
 #include <asm/mach-ath79/ar71xx_regs.h>
+#include <linux/platform/ar934x_nfc.h>
 
 #include "common.h"
 #include "dev-eth.h"
@@ -239,6 +240,7 @@ void __init rb95x_nand_init(void)
 {
 	gpio_request_one(RB95X_GPIO_NAND_NCE, GPIOF_OUT_INIT_HIGH, "NAND nCE");
 
+	ath79_nfc_set_ecc_mode(AR934X_NFC_ECC_SOFT_BCH);
 	ath79_nfc_set_scan_fixup(rb95x_nand_scan_fixup);
 	ath79_nfc_set_parts(rb95x_nand_partitions,
 			    ARRAY_SIZE(rb95x_nand_partitions));
